lib/ansible/plugins/action/nxos.py

```diff
@@ -75,32 +75,11 @@ def run(self, tmp=None, task_vars=None):
                 task_vars['ansible_socket'] = socket_path
 
             else:
-                provider['transport'] = 'nxapi'
-                if provider.get('host') is None:
-                    provider['host'] = self._play_context.remote_addr
-
-                if provider.get('port') is None:
-                    if provider.get('use_ssl'):
-                        provider['port'] = 443
-                    else:
-                        provider['port'] = 80
-
-                if provider.get('timeout') is None:
-                    provider['timeout'] = C.PERSISTENT_COMMAND_TIMEOUT
-
-                if provider.get('username') is None:
-                    provider['username'] = self._play_context.connection_user
-
-                if provider.get('password') is None:
-                    provider['password'] = self._play_context.password
-
-                if provider.get('use_ssl') is None:
-                    provider['use_ssl'] = False
-
-                if provider.get('validate_certs') is None:
-                    provider['validate_certs'] = True
-
-                self._task.args['provider'] = provider
+                obj = {}
+                obj['remote_addr'] = self._play_context.remote_addr
+                obj['connection_user'] = self._play_context.connection_user
+                obj['password'] = self._play_context.password
```

Comment:
Can setting above three keys moved in  `nxapi_implementation()`

Reply:
We do not need to pass `self` for `staticmethod`. 
I do not think that we should tie this method to the specific class for reusability that is why using `staticmethod`.